src/decode.c

```diff
@@ -827,6 +827,8 @@ static inline bool op_system(rv_insn_t *ir, const uint32_t insn)
         case 0x105: /* WFI: Wait for Interrupt */
         case 0x002: /* URET: return from traps in U-mode */
         case 0x102: /* SRET: return from traps in S-mode */
+            ir->opcode = rv_insn_sret;
```

Comment:
Did you treat WFI (Wait for Interrupt), URET (User-mode Return from Exception), and SRET (Supervisor-mode Return from Exception) instructions as the SRET instruction?

Reply:
Nope, I have reorder the switch cases to avoid this happens.